### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by aarnott50**
Some code cleanup that I thought would be helpful after doing a cursory scan of the project. My IDE removed a lot of trailing whitespace, so the changes look a bit more substantial than they actually are.

### CODE REVIEW

File: ashley/src/com/badlogic/ashley/signals/Signal.java [hunk 1]
```diff
@@ -56,9 +56,8 @@ public void removeAllListeners () {
 	 * @param object The object to send off
 	 */
 	public void dispatch (T object) {
-		Object[] items = listeners.begin();
-		for (int i = 0, n = listeners.size; i < n; i++) {
-			Listener<T> listener = (Listener<T>)items[i];
+		final Listener<T>[] items = listeners.begin();
```
Comment (aarnott50) [inline ashley/src/com/badlogic/ashley/signals/Signal.java]: Arrays of generics is a bad idea and should generally be avoided. In this case, it looks like the base libgdx library is returning it, however, so there isn't much that can be done here.

See: http://www.angelikalanger.com/Articles/Papers/JavaGenerics/ArraysInJavaGenerics.htm (specifically the Parameterized Types As Array Elements section), if you are interested :).

**Comment by saltares** [comment]
Thanks for the PR. It looks good, thanks for the array of generics heads up. Just to make extra sure (although it looks like they would) have you checked the unit tests are still passing?

**Comment by aarnott50** [comment]
All of the unit tests pass, yes. I ran gradle test explicitly, but it should just happen as part of building it, correct?

**Comment by saltares** [comment]
afaik you need to run them explicitly, but since you did, I'll just merge. Thanks!

File: ashley/src/com/badlogic/ashley/signals/Signal.java (hunk 1 above)
Comment (saltares) [inline ashley/src/com/badlogic/ashley/signals/Signal.java]: Actually, this broke the build.

http://libgdx.badlogicgames.com:8080/job/ashley/103/console

**Comment by saltares** [comment]
You cannot have run the test successfully, because most of them just crash as a cause of these changes. Reverting the array of generics for now.

**Pull request merged**